api/src/db/models/entity_models.py

```diff
@@ -0,0 +1,69 @@
+import uuid
+from datetime import date
+
+from sqlalchemy import ForeignKey, Integer, Text
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from src.adapters.db.type_decorators.postgres_type_decorators import LookupColumn
+from src.db.models.base import ApiSchemaTable, TimestampMixin
+from src.db.models.lookup_models import LkSamGovImportType
+
+
+class SamGovEntity(ApiSchemaTable, TimestampMixin):
+    __tablename__ = "sam_gov_entity"
+
+    sam_gov_entity_id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True, default=uuid.uuid4)
+    uei: Mapped[str] = mapped_column(Text, nullable=False, index=True)
+    legal_business_name: Mapped[str]
+    expiration_date: Mapped[date]
+    ebiz_poc_email: Mapped[str]
+    ebiz_poc_first_name: Mapped[str]
+    ebiz_poc_last_name: Mapped[str]
+    has_debt_subject_to_offset: Mapped[bool | None]
+    has_exclusion_status: Mapped[bool | None]
+    eft_indicator: Mapped[str | None]
+
+    # Relationships
+    import_records: Mapped[list["SamGovEntityImportType"]] = relationship(
+        back_populates="sam_gov_entity", cascade="all, delete-orphan"
+    )
+    organizations: Mapped[list["Organization"]] = relationship(
+        back_populates="sam_gov_entity", cascade="all, delete-orphan"
```

Comment:
I think we wouldn't want cascade delete here - I don't think we'll ever really delete an entity record (probably soft deletes?), but to be safe could we not put this? Deleting an organization once we start adding to it would be really bad.